src/bijectors/corr.jl

```diff
@@ -173,11 +173,30 @@ end
 
 inverse(::typeof(vec_to_triu1)) = triu1_to_vec
 
+function vec_to_triu1_row_index(idx)
+    # Assumes that vector was saved in a column-major order
+    # and that vector is one-based indexed.
+    M = _triu1_dim_from_length(idx - 1)
+    return idx - (M*(M-1) ÷ 2)
+end
+
 """
     VecCorrBijector <: Bijector
 
-Similar to `CorrBijector`, but correlation matrix to a vector,
-and its inverse transforms vector to a correlation matrix.
+A bijector to transform either a correlation matrix or a Cholesky factor of a correlation matrix
+to an unconstrained vector. 
+
+# Fields
+- mode :`Symbol`. Controls the inverse tranformation :
+    - if `mode === :C` returns a correlation matrix
```

Comment:
That is, make the cholesky version into a separate type, e.g. `VecCholCorrBijector`. This will avoid the type-instabilities + moves the conditional handling you have in some functions into multiple dispatch instead.

Reply:
@harisorgn Any updates on this?:)

Reply:
I was completely off last week. Agree with splitting/specialising the structures, will implement it this week!

Reply:
Ah, no worries! Sweet!